### PR TITLE
Add support for configuring max dimension

### LINKED ISSUE #221: Size limited to 8192x8192
On larger images I get an error:
```
image dimensions cannot be larger than 8192x8192, received image dimensions 12560x17552
```

Would it be possible to increase the maximum size to let's say 32768x32768 or ist this a limitation of the JPEG container?

### PR DESCRIPTION
**Pull request by ram-mohan**
uhdr library defaults max supported dimension to 8k resolution. For handling images with resolution exceeding 8192x8192, reconfigure the library with option UHDR_MAX_DIMENSION

fixes #221

Build: cmake -DUHDR_MAX_DIMENSION=16384 ../

### CODE REVIEW

File: lib/src/ultrahdr_api.cpp [hunk 1]
```diff
@@ -26,6 +26,38 @@
...
+const int kMapCompressQualityDefault = 85;
+
+// Gain map calculation
+const bool kUseMultiChannelGainMapDefault = false;
+// Default gamma value for gain map
+const float kGainMapGammaDefault = 1.0f;
+
+// The current JPEGR version that we encode to
+const char* const kGainMapVersion = "1.0";
+const char* const kJpegrVersion = "1.0";
```
Comment (DichenZhang1) [inline lib/src/ultrahdr_api.cpp:43]: This is JPEG/R related, I prefer to keep it under jpegr.h

Comment (ram-mohan) [reply lib/src/ultrahdr_api.cpp]: Done

File: lib/src/ultrahdr_api.cpp [hunk 2]
```diff
@@ -26,6 +26,38 @@
 
 namespace ultrahdr {
 
+// Default configurations
+// Map is quarter res / sixteenth size
+const size_t kMapDimensionScaleFactorDefault = 4;
```
[the remark this answers is not in the document]
Comment (DichenZhang1) [reply lib/src/ultrahdr_api.cpp:39]: Same comment.

[the remark this answers is not in the document]
Comment (ram-mohan) [reply lib/src/ultrahdr_api.cpp]: I have addressed the open comments, dichen.

**Review by DichenZhang1 — APPROVED** [review]

**Pull request merged**